host_source.go

```diff
@@ -635,6 +635,11 @@ func (r *ringDescriber) getClusterPeerInfo(localHost *HostInfo) ([]*HostInfo, er
 		return nil, errNoControl
 	}
 
+	// Check if host lookup is disabled
+	if r.session.cfg.DisableHostLookup {
+		return []*HostInfo{}, nil
+	}
```

Comment:
Instead of adding this, I think we should make the `ringRefresher` do a NOOP when this flag is set. `getClusterPeerInfo` should only be about retriving the peers info from the server so it should be up to the caller whether this should be done or not.

If we make the `func refreshRing(r *ringDescriber) error` function do nothing when the flag is set then it should solve all of the issues related to the driver not respecting the flag.